src/CommandExecutor.ts

```diff
@@ -51,6 +51,12 @@ class BuiltInCommandExecutionStrategy extends CommandExecutionStrategy {
 class SystemFileExecutionStrategy extends CommandExecutionStrategy {
     startExecution() {
         return new Promise((resolve, reject) => {
```

Comment:
I've refactored the way we choose a strategy to make it easier to extend. Sorry if it created conflicts.